airflow/providers/google/cloud/sensors/dataflow.py

```diff
@@ -357,4 +500,45 @@ def poke(self, context: Context) -> bool:
             location=self.location,
         )
 
-        return self.callback(result)
+        return result if self.callback is None else self.callback(result)
+
+    def execute(self, context: Context) -> Any:
+        """Airflow runs this method on the worker and defers using the trigger."""
+        if not self.deferrable:
+            super().execute(context)
+        else:
```

Comment:
```suggestion
        elif not self.poke(context=context):
```

Reply:
@Lee-W we have already had this discussion earlier. I argued that it would be better not to do this for 3 out of 4 sensors, because of the user-defined `callback` parameter which could return arbitrary data types.

I implemented this suggestion for `DataflowJobStatusSensor` which does not have this parameter.

Reply:
@Lee-W Please check my comment here. It should address your suggestions below.

Reply:
Yep, makes sense